src/monster.cpp

```diff
@@ -3118,9 +3118,15 @@ void monster::on_load()
     const int heal_amount = roll_remainder( regen * to_turns<int>( dt ) );
     const int healed = heal( heal_amount );
     int healed_speed = 0;
-    if( healed < heal_amount && get_speed_base() < type->speed ) {
-        int old_speed = get_speed_base();
-        set_speed_base( std::min( get_speed_base() + heal_amount - healed, type->speed ) );
+    if( get_speed_base() < type->speed ) {
+        const int old_speed = get_speed_base();
+        if( hp >= type->hp ) {
+            set_speed_base( type->speed );
+        } else {
+            const int speed_delta = std::max( healed * type->speed / type->hp, 1 );
+            set_speed_base( std::min( old_speed + speed_delta, type->speed ) );
+                                      type->speed ) );
```

Comment:
```suggestion
            set_speed_base( std::min( old_speed + speed_delta, type->speed ) );
```

Something got messed up here. Let's remove the junk.

Reply:
Github is insisting that there's no change there - doing a pull to my local machine to check.